src/audio/component.c

```diff
@@ -194,8 +197,7 @@ int comp_set_state(struct comp_dev *dev, int cmd)
 
 void sys_comp_init(void)
 {
-	cd = rzalloc(RZONE_SYS, SOF_MEM_CAPS_RAM, sizeof(*cd));
-	list_init(&cd->list);
+	list_init(&cd.list);
```

Comment:
I think you have to write back cache here, otherwise you invalidate it next time you enter *_register() or *_unregister() above